src/command_modules/azure-cli-role/azure/cli/command_modules/role/_params.py

```diff
@@ -15,9 +15,23 @@
 register_cli_argument('ad', 'spn', help='service principal name')
 register_cli_argument('ad', 'upn', help='user principal name, e.g. john.doe@contoso.com')
 register_cli_argument('ad', 'query_filter', options_list=('--filter',), help='OData filter')
-register_cli_argument('role assignment', 'role_assignment_name',
-                      options_list=('--role-assignment-name', '-n'))
-register_cli_argument('role assignment', 'role', help='role name or id')
 register_cli_argument('ad user', 'mail_nickname',
                       help='mail alias. Defaults to user principal name')
 register_cli_argument('ad user', 'force_change_password_next_login', action='store_true')
+register_cli_argument('role assignment', 'role_assignment_name',
+                      options_list=('--role-assignment-name', '-n'))
+register_cli_argument('role assignment', 'role', help='role name or id')
+register_cli_argument('role assignment', 'show_all', options_list=('--all',), action='store_true',
+                      help='show all assignments under the current subscription')
+register_cli_argument('role assignment', 'include_inherited', action='store_true',
+                      help='include assignments applied on parent scopes')
+register_cli_argument('role assignment', 'assignee', help='represent a user, group, or service principal. supported format: object id, user sign-in name, or service principal name')
+register_cli_argument('role assignment', 'ids', nargs='+', help='space separated role assignment ids')
+register_cli_argument('role', 'role_id', help='the role id')
+register_cli_argument('role', 'resource_group_name', options_list=('--resource-group', '-g'),
+                      help='use it only if the role or assignment was added at the level of a resource group')
+register_cli_argument('role', 'resource_id', help='use it only if the role or assignment was added at the level of a resource')
+register_cli_argument('role', 'custom_role_only', action='store_true', help='custom roles only(vs. build-in ones)')
+register_cli_argument('role', 'role_definition', help="json formatted content which defines the new role. run 'show-create-template' to get samples")
```

Comment:
>help="json formatted content which defines the new role. run 'show-create-template' to get samples [](start = 49, length = 98)

Should be more clear. It is either the path to the JSON definition file or the raw JSON definition itself.